[Review]

lang92 opened the pull request:
리뷰 PR입니다. 머지하시면 안되고, 각자 작업하신 내용에 대한 리뷰를 반영해 주세요!

GiwanKK committed:
[etc] common.scss 추가

GiwanKK committed:
[etc] common.scss 추가

syon013 committed:
Merge pull request #1 from wecode-bootcamp-korea/etc/common

Etc/common

GiwanKK committed:
[style] nav 이동 및 토큰값 로직 + UI 구현

GiwanKK committed:
[style] 메인페이지 UI완료

syon013 committed:
Merge pull request #3 from wecode-bootcamp-korea/component/main

Component/main

syon013 committed:
[feat] button component 추가

syon013 committed:
[feat] button component / props에 onClick 추가

syon013 committed:
[style] input component 추가

syon013 committed:
[etc] img 이름 수정, url 주소 수정

syon013 committed:
[etc] cart.img 이름 수정

GiwanKK committed:
Merge pull request #4 from wecode-bootcamp-korea/component/button

[feat] button component 추가

syon013 committed:
[style] Login component 추가

GiwanKK committed:
[style] 제품리스트 UI 구현

syon013 committed:
[etc] common.scss legend 속성 추가

syon013 committed:
Merge pull request #6 from wecode-bootcamp-korea/component/products

[style] 제품리스트 UI 구현

syon013 committed:
Merge branch 'master' of https://github.com/wecode-bootcamp-korea/49-2nd-TeaTime-frontend into component/login

syon013 committed:
[feat][style] Login UI 및 기능 구현

syon013 committed:
[etc] NAV 이동 경로 추가

syon013 committed:
[feat][style] 로그인 관련 버튼 UI 및 기능 추가

syon013 committed:
[feat][style] 로그인 관련 UI 및 기능 추가

GiwanKK committed:
Merge pull request #7 from wecode-bootcamp-korea/component/login

Component/login

syon013 committed:
[style] signup UI 구현 시작

syon013 committed:
Merge branch 'master' of https://github.com/wecode-bootcamp-korea/49-2nd-TeaTime-frontend into component/signup

lang92 commented on src/pages/Login/Component/MainForm/MainForm.js:
```diff
@@ -0,0 +1,134 @@
+    e.preventDefault();
+  };
+
+  const loginValidation = e => {
+    const idRegExp = /^[A-Za-z0-9]{6,}$/;
+    const passwordRegEx =
+      /^(?=.*[a-zA-Z])(?=.*[!@#$%^*+=-])(?=.*[0-9]).{8,20}$/;
+
+    if (!idRegExp.test(userId)) {
+      // alert('6글자 이상 소문자와 숫자로 혼합된 아이디를 입력해주세요.');
+      return false;
+    } else if (!passwordRegEx.test(userPw)) {
+      // alert(
+      //   '비밀번호는 영문, 숫자, 특수기호를 조합하여 8~20글자 이상 입력해주세요.',
+      // );
+      return false;
+    } else {
+      return true;
+    }
+  };
```
alert를 출력하지 않는다면, 이렇게 변수로 사용하시면 될 것 같습니다.
```suggestion
  const idRegExp = /^[A-Za-z0-9]{6,}$/;
  const passwordRegEx = /^(?=.*[a-zA-Z])(?=.*[!@#$%^*+=-])(?=.*[0-9]).{8,20}$/;
  const isLoginInputValid = idRegExp.test(userId) && passwordRegEx.test(userPw);
```

lang92 commented on src/pages/Login/Component/MainForm/MainForm.js:
```diff
@@ -0,0 +1,134 @@
+            shape="fill"
+            color="primary"
+            fullWidth="true"
+            disabled={!loginValidation() ? true : false}
```
- 위 리뷰 내용 참고
- falsy일 때 true, 반대일 때 false라면 조건의 반댓값으로 넣어주면 됨

```suggestion
            disabled={!isLoginInputValid}
```

lang92 commented on src/pages/Login/Login.scss:
```diff
@@ -0,0 +1,7 @@
+.nav + .login {
+  margin-top: 80px;
+}
+
+.login {
+  width: 100%;
+}
```
```suggestion
.login {
  width: 100%;
  margin-top: 80px;
}
```

lang92 commented on src/pages/Main/Components/BestList.js:
목 데이터 + map 메서드로 작성하시면 더 좋을 것 같습니다.

GiwanKK replied:
일단 UI먼저 적용하느라 기능적인 부분 할 때 수정하겠습니다.

lang92 commented on src/pages/Main/Main.scss:
```diff
@@ -0,0 +1,149 @@
+      height: 550px;
+    }
+  }
+  .trboxBtnBox,
+  .prdBtnBox {
+    margin-bottom: 28px;
+    text-align: center;
+    max-width: 1240px;
+    padding: 0 20px;
+    margin-left: auto;
+    margin-right: auto;
+  }
+  .trBoxMore,
+  .bestPrdMore {
```
동일한 스타일을 가진다면, className을 따로 부여할 필요 없이 포괄적인 이름을 가진 하나의 className으로 처리하시면 됩니다.

lang92 commented on src/pages/Products/Components/Contents.js:
```diff
@@ -0,0 +1,272 @@
+      <p className="pageLocation">{props.titleText}</p>
+      <div className="sortBar">
+        <ul>
+          {props.productSort.map(item => (
```
productSort로 들어오는 prop은 이 곳에서만 사용되는 상수 데이터인 것 같은데, props로 받지 말고 이 컴포넌트에서 import하여 사용하시면 됩니다.

GiwanKK replied:
수정했습니다

lang92 commented on src/pages/Products/Components/Contents.js:
```diff
@@ -0,0 +1,272 @@
+        <ul>
+          {props.productSort.map(item => (
+            <li key={item.text}>
+              <a
+                className={props.sort === item.value && 'on'}
+                id={item.value}
+                onClick={e => {
+                  props.handlerSort(e);
+                }}
+              >
+                {item.text}
+              </a>
```
a 태그는 외부 페이지로 이동할 때 사용하는 태그입니다. 다른 태그로 사용해 주세요!

lang92 commented on src/pages/Products/Components/Contents.js:
```diff
@@ -0,0 +1,272 @@
+      </div>
+      <div className="productList">
+        <div className="listWrapper">
+          {/* 여기서부터 반복 */}
+          <div className="item">
+            <div className="thumb">
+              <a>
+                <img src="https://image.osulloc.com/upload/kr/ko/adminImage/JY/QU/304_20210825170710069ZN.png"></img>
+              </a>
+              <a className="basket">
+                <button>장바구니</button>
+              </a>
+            </div>
+            <p className="prdName">
+              <a>러블리 티 박스</a>
+            </p>
+            <span className="prdPrice">
+              <p className="">
+                <strong>22,400원</strong>
+                <br />
+                <del>28,000원</del>
+                <em>(20%)</em>
+              </p>
+            </span>
+          </div>
+          {/* 여기까지 반복 */}
```
ㅎㅎ.. 반복되는 부분은 상수 데이터와 map 메서드로!

lang92 commented on src/styles/common.scss:
```diff
@@ -3,3 +3,66 @@
+  z-index: -1;
+}
+
+@mixin center {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
```
mixin.scss에 있기 때문에, 지워 주세요!

lang92 commented on src/pages/Products/Products.js:
```diff
@@ -0,0 +1,72 @@
+  const [page, setPage] = useState(1);
+
+  useEffect(() => {
+    if (searchParams.get('category'))
+      setCategory(parseInt(searchParams.get('category')));
+    if (searchParams.get('sort')) setSort(parseInt(searchParams.get('sort')));
+    if (searchParams.get('page')) setPage(parseInt(searchParams.get('page')));
```
searchParams의 값으로 state를 관리할 필요 없이, searchParams만 사용하시면 됩니다.
state는 전부 지우셔도 될 것 같네요!

GiwanKK replied:
일반 페이지 이동에서는 searchParams의 값이 없고 그에 따라 초깃값에 대한 버튼들이 불이 들어와야 하는 구조에서 초깃값 세팅을 위해 state 값 을 사용하는 로직을 썼는데 잘못된 선택인 건가요??

lang92 reviewed (commented):
리뷰는 반복해서 달아드리지 않았으니, 본인이 작업한 내용이 아니라도 모두 읽어 주세요!
그리고 나서 각자 작업한 내용에 반영할 내용 반영해 주시면 됩니다.

또 제 개인적인 기준에서는 전체적으로 컴포넌트를 너무 잘게 쪼개셨는데, 이 부분은 각자 작업해 보시면서 느끼신 불편함이나 이점 등에 대해 잘 생각해 보시면 좋을 것 같습니다.

GiwanKK committed:
Merge pull request #5 from wecode-bootcamp-korea/component/input

Component/input 추가

GiwanKK commented on src/Component/Nav/Nav.js:
```diff
@@ -0,0 +1,126 @@
+} from '../../data/ComponentData';
+
+const Nav = () => {
+  const [style, setStyle] = useState({ display: 'none' });
```
수정했습니다

GiwanKK commented on src/Component/Nav/Nav.js:
```diff
@@ -0,0 +1,126 @@
+
+const Nav = () => {
+  const [style, setStyle] = useState({ display: 'none' });
+  let isLoginStatus = localStorage.getItem('token');
```
수정했습니다

GiwanKK commented on src/Component/Nav/Nav.js:
```diff
@@ -0,0 +1,126 @@
+  let isLoginStatus = localStorage.getItem('token');
+  const nav = useNavigate();
+
+  const dimBgAble = (state, key) => {
+    if (key === 0) {
+      setStyle({ display: state });
+    }
+  };
```
수정했습니다

GiwanKK commented on src/Component/Nav/Nav.js:
```diff
@@ -0,0 +1,126 @@
+const Nav = () => {
+  const [style, setStyle] = useState({ display: 'none' });
+  let isLoginStatus = localStorage.getItem('token');
+  const nav = useNavigate();
```
수정했습니다

GiwanKK commented on src/Component/Nav/Nav.js:
```diff
@@ -0,0 +1,126 @@
+                    <span className="cartNum">0</span>
+                  </button>
+                </li>
+                {!isLoginStatus ? (
+                  <li className="item">
+                    <Link to="/login">로그인</Link>
+                  </li>
+                ) : (
+                  <li className="item">
+                    <Link to="/">마이페이지</Link>
+                  </li>
+                )}
+                {!isLoginStatus ? (
+                  <li className="item">
+                    <Link to="/signup">회원가입</Link>
+                  </li>
+                ) : (
+                  <li className="item">
+                    <Link onClick={logOut}>로그아웃</Link>
+                  </li>
+                )}
```
수정했습니다

GiwanKK commented on src/Component/Nav/Nav.js:
```diff
@@ -0,0 +1,126 @@
+                      onMouseLeave={() => dimBgAble('none', index)}
+                    >
+                      <Link to={item.url}>{item.text}</Link>
+                      {item.child && (
+                        <div className="navListDept2Box">
+                          <div className="wrapper">
+                            <div className="flexBox">
+                              <ul className="navListDept2">
+                                <li className="itemDept2">
+                                  <Link to="/">베스트</Link>
+                                  <ul className="navListDept3">
+                                    {BEST_PRODUCT_MENU.map(item => (
+                                      <li key={item.text} className="itemDept3">
+                                        <Link to={item.url}>{item.text}</Link>
+                                      </li>
+                                    ))}
+                                  </ul>
+                                </li>
+                                <li className="itemDept2">
+                                  <Link to="/products">티 제품</Link>
+                                  <ul className="navListDept3">
+                                    {PRODUCT_MENU.map(item => (
+                                      <li key={item.text} className="itemDept3">
+                                        <Link to={item.url}>{item.text}</Link>
+                                      </li>
+                                    ))}
+                                  </ul>
+                                </li>
+                              </ul>
+                            </div>
+                          </div>
+                        </div>
+                      )}
```
이해가 잘 가지 않습니다ㅠ

GiwanKK commented on src/Component/Nav/Nav.scss:
```diff
@@ -0,0 +1,187 @@
+  font-weight: 400;
+  font-size: 13px;
+
+  * {
+    box-sizing: border-box;
+    margin: 0;
+    padding: 0;
+  }
```
수정했습니다

GiwanKK commented on src/Component/Nav/Nav.scss:
```diff
@@ -0,0 +1,187 @@
+    &:hover {
+      color: green;
+    }
+  }
+  .dimBg {
+    position: fixed;
+    top: 0;
+    left: 0;
+    width: 100vw;
+    height: 100vh;
+    background-color: rgba(0, 0, 0, 0.6);
+    z-index: 99;
+  }
+  .menuBox {
+    position: fixed;
+    top: 0;
+    left: 0;
+    width: 100%;
+    z-index: 100;
+    background: white;
+  }
+  .wrapper {
+    padding: 0 10px;
+    max-width: 1540px;
+    margin-left: auto;
+    margin-right: auto;
+  }
```
수정했습니다

syon013 committed:
[etc] common 내용 삭제 (리뷰 반영)

syon013 committed:
[feat][style] SignUp UI 및 기능 구현 진행 중

syon013 committed:
Merge branch 'master' of https://github.com/wecode-bootcamp-korea/49-2nd-TeaTime-frontend into component/signup

syon013 committed:
[etc] 중복되는 버튼 컴포넌트 삭제 (리뷰 반영)

syon013 committed:
[style] 회원가입 UI 완성

syon013 committed:
[feat] input CheckBox disabled 색상 변경 기능 추가

syon013 committed:
[etc] props 내용 중 className 관련 내용 수정 (리뷰 반영)

syon013 committed:
[style] 회원가입 이동 버튼 UI 수정

syon013 committed:
[feat] SignUp용 Input 기능 추가

syon013 committed:
[style] Login용 Form UI 수정

syon013 committed:
[feat] SignUp 기능 구현 완료

syon013 committed:
[etc] 리뷰 반영 파일 삭제 및 필요없는 주석 제거

syon013 commented on src/Component/Button/Button.js:
```diff
@@ -0,0 +1,38 @@
+import './Button.scss';
+
+const Button = props => {
+  const {
+    className = 'btn',
+    type = 'button',
+    scale,
+    shape,
+    item,
+    disabled,
+    style,
+    fullWidth = false,
+    color,
+    onClick,
+    text,
+    children,
+  } = props;
+  return (
+    <button
+      className={`btn ${className}`}
+      type={type}
+      scale={scale}
+      shape={shape}
+      item={item}
+      disabled={disabled}
+      style={style}
+      fullWidth={fullWidth}
+      color={color}
+      onClick={onClick}
+      text={text}
+    >
+      {children}
+    </button>
+  );
```
수정했습니다.

syon013 commented on src/Component/Button/Button.scss:
```diff
@@ -0,0 +1,79 @@
+// props 정리
+// className: 버튼의 클래스명
+// type: 버튼의 타입 (button, submit)
+// scale: 버튼의 크기(small, medium, large)
+// fullWidth : width 100% 여부
+// disabled: 버튼의 비활성화 여부
+// shape: 버튼의 모양(fill, outline, tap)
+// img : 버튼의 이미지 (close, ...)
+// color = primary, secondary
+// $colorA: #5c95f0, $colorB: #313131,
+// $colorC: #ffffff, $colorD: #000000,
+// $colorE: #6c801a, $colorF: #666,
+// $colorG: #ddd,
```
JSDoc 을 사용하는 방법이 잘 이해가 되질 않습니다 ...... 그냥 README에 올려도되나요 ..?

syon013 commented on src/Component/ButtonWrap/ButtonWrap.js:
```diff
@@ -0,0 +1,25 @@
+import Button from '../Button/Button';
+import './ButtonWrap.scss';
+
+const ButtonWrap = props => {
+  const { className, type, scale, text, onClick, disabled, shape, children } =
+    props;
+  return (
+    <div className={`btnWrap ${className}`}>
+      <Button
+        className={className}
+        type={type}
+        scale={scale}
+        text={text}
+        onClick={onClick}
+        disabled={disabled}
+        shape={shape}
+      >
+        {children}
+      </Button>
+    </div>
+  );
+};
```
삭제했습니다.

syon013 commented on src/Component/IconBtnWrap/IconBtnWrap.js:
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import './IconBtnWrap.scss';
+import IconButton from '../IconButton/IconButton';
```
해당 폴더 삭제했습니다.

syon013 commented on src/Component/IconBtnWrap/IconBtnWrap.scss:
```diff
@@ -0,0 +1,14 @@
+.iconBtnWrap {
+  display: flex;
+  flex-direction: column;
+  font-size: 15px;
+
+  & span {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    margin-top: 55px;
+    line-height: 1.23;
+    color: #252525;
+  }
+}
```
폴더 삭제했습니다.

syon013 committed:
[style] Mypage UI 구현중

syon013 committed:
[feat] Warning 제거

1. Link 태그는 a 태그로 Converting 되기 때문에 children의 a 태그 제거
2. html custom attr(속성) 는 data- 로 시작 해야한다
fullWidth -> data-fullWidth 로 변경

syon013 committed:
[refactor] btn className 중복 입력 제거

### 원인
Button 컴포넌트와, IconButton 컴포넌트의 className이  btn, iconBtn 이 중복으로 들어가는 현상

### 조치
className 의 default 값을 제거하고
className 이 있을 경우 :: `btn ${className}`
className 이 없을 경우 :: 'btn'
형태로 넣을 수 있도록 변경
IconButton 도 동일하게 변경

syon013 committed:
[style] 카카오 버튼 임시 주석처리

### 이유
backend 미개발로 인해 임시 주석처리하여 화면상에 보이지 않도록 변경

syon013 committed:
[etc] JsDoc 적용

Button 에 JsDoc 적용시키기

GiwanKK committed:
Merge pull request #9 from wecode-bootcamp-korea/component/signup

Component/signup

GiwanKK committed:
[feat] paging 기능 구현

GiwanKK committed:
[feat] paging 기능 구현 props 추가

syon013 committed:
Merge pull request #10 from wecode-bootcamp-korea/component/paging

Component/paging

GiwanKK committed:
[style] 상품 상세 UI 구현 중

syon013 committed:
[feat] Count Component 추가

### Count 관련 컴포넌트 UI 구현 완료

syon013 committed:
[feat] Count Component 관련 Props 추가

syon013 committed:
[etc] Common.scss 내용 추가

1. img 태그 관련 100% 속성 추가 (추후 논의 후 존속/삭제 여부 결정)

syon013 committed:
[style] MyPage UI 구현 진행 중

1. master 추가 사항 머지 전 임시 커밋 진행
2. 최근 구매 목록 UI까지 구현 완료 (페이징 관련 UI는 미진행)

syon013 committed:
Merge branch 'master' of https://github.com/wecode-bootcamp-korea/49-2nd-TeaTime-frontend into component/mypage

syon013 committed:
[style] Chip Component 추가

기본 배송지 및 기본 배송지 설정 UI 인 Chip Component UI 구현 완료

syon013 committed:
[feat] myPage Tab 기능 구현

- 회원 정보, 최근 구매목록, 배송지 설정 페이지 랜더링 관련 TAB 기능 구현 완료

syon013 committed:
[feat] TAB Component 추가 및 기능 구현

마이페이지 TAB 기능 구현 완료
### 회원 정보, 최근 구매 목록, 배송지 정보 페이지 랜더링

GiwanKK committed:
[style] 버튼 컴포넌트 추가

syon013 committed:
[style] 회원정보 UI 구현

### 회원 정보 UI 구현 완료

syon013 committed:
[style] 마이페이지 '배송지 정보' UI 구현

1. 마이페이지 배송지 정보 UI 구현 완료
2. 배송지 추가 버튼 onClick 시 배송지 추가 관련 Form UI 구현 완료
3. 배송지 추가 관련 Button, Input Customer 추가

syon013 committed:
[etx] TABS 상수데이터명 수정

syon013 committed:
[style] 마이페이지 '최근 주문내역' UI 구현

1. 날짜별 정렬 목데이터 이용하여 구현되도록 UI 구현 (추가 사항이기 때문에 추후 진행할 시 백엔드와 논의 필요)

2. 최근 주문내역 주문 진행 UI 구현 완료

syon013 committed:
[style] 마이페이지 회원정보, 최근 주문내역, 배송지 정보 관련 UI 구현 완료

### 목데이터 및 하드 코딩으로 우선 구현 (추가 사항 이기 때문에 추후 기능 개발 예정)

syon013 committed:
[feat] TAB 선택 기능 구현

### TAB 선택 시 bottom에 border 생기도록 구현 완료

syon013 committed:
[style] 회원가입 아이디 중복체크 버튼 UI 추가

### 버튼 UI 구현

GiwanKK committed:
[style] 상품 디테일 페이지 UI 리뷰 진행 중

GiwanKK committed:
Merge pull request #12 from wecode-bootcamp-korea/component/mypage

Component/mypage

syon013 committed:
[style] payment Component 생성

syon013 committed:
Merge branch 'master' into component/product

syon013 committed:
Merge pull request #11 from wecode-bootcamp-korea/component/product

[style] 상품 상세 UI 구현 중

syon013 committed:
[etc] payment Router 경로 추가

GiwanKK committed:
[etc] 충돌 에러 수정

GiwanKK committed:
Merge pull request #13 from wecode-bootcamp-korea/component/signup

[style] 회원가입 아이디 중복체크 버튼 UI 추가

GiwanKK committed:
Merge branch 'master' into component/payment

GiwanKK committed:
Merge pull request #14 from wecode-bootcamp-korea/component/payment

Component/payment

syon013 committed:
Merge pull request #15 from wecode-bootcamp-korea/component/product

[etc] 충돌 에러 수정

syon013 committed:
[etc] 베스트 상품 UI 생성을 위한 임시 커밋

syon013 committed:
[etc] Best Component 목데이터 생성 및 이미지 등록

1. best 페이지에 들어갈 데이터 관련하여 목데이터 생성

2. 홈페이지 기준 베스트 아이템은 12개이며, 12개의 이미지 및 페이지 배너 이미지 등록

syon013 committed:
[feat] best 페이지 라우터 및 Nav 경로 지정

1. Router/Nav best 페이지 경로 지정

syon013 committed:
[style] 홈페이지에 들어가는 Chip UI 구현

### 기능은 없으며, 순수 UI만 등록되어 있음.

syon013 committed:
[style] ImgBanner Component 생성

1. ImgBanner Component 화 진행 UI 생성 완료
2. 해당 Component Best.js 적용 완료

syon013 committed:
[style] BestList Component 화 완료

1. Best Component 에서 코드 분리 완료
2. 추후 상세페이지 / 장바구니 페이지 구현 완료 시 Router 경로 수정 필요
3. BackEnd Data 나오면 붙여서 확인 필요

syon013 committed:
[style] Best Component UI 구현 완료

1. 추후 상세 페이지 / 장바구니 페이지 구현 완료 시 링크 수정 필요

2. BackEnd Data 나오면 해당 Data들 붙여서 정상적으로 받아오는지 확인 필요

syon013 committed:
[style] BestList UI 수정

1. 각각 개별로 margin : 0 auto; 가 걸려있어 데이터가 정확하게 4개씩 들어올 떄는 문제가 없으나 2개가 들어오면 이상하게 정렬되는 이슈가 있어 해당 사항 수정

2. 기존 bestListWrap 에 있던 margin : 0 auto; 삭제 후 width: 25%, padding : 1% 로 수정

GiwanKK committed:
Merge pull request #16 from wecode-bootcamp-korea/component/best

Component/best

syon013 committed:
[feat] Best Page 기능 구현

1. gift 관련 목업데이터 생성

2. Chip.scss 수정
 - user-select : none; 추가하여 더블클릭 되도 드래그 되지 않도록 수정

3. best Chip 기능 구현하여 Chip의 상태가 바뀔 때마다 해당 상품 List 바뀌도록 기능 구현 완료

4. BestList에 따라 상품 List 보여지도록 구현 및 가격에 자릿수에 맞는 ( , ) 찍히도록 구현 완료
4-1. 상세페이지 및 장바구니는 아직 미구현 상태라서 onClick 시 메인페이지 이동되도록만 구현

5. GiftBest Component를 만들어 best와 같은 기능 구현되도록 구현

syon013 committed:
[style] ImgBanner Style 수정

1. text-shadow 속성 변경

2. user-select : none; 으로 더블 클릭 되도 이미지 배너 관련 내용 선택 되지 않도록 수정

syon013 committed:
[feat] 배송지 추가 관련 API 구현

1. 버튼에 연결만 해놓은 상태이며 검색 완료 시 INPUT에 데이터를 넣는 쪽은 구현하지 않은 상태임.

 - 추가 구현을 진행하면 구현 해볼 예정임.

GiwanKK committed:
[style] 제품 상세 UI구현 완료

GiwanKK committed:
[style] 제품 상세 UI구현 완료

GiwanKK committed:
[etc] 각 페이지 장바구니 버튼 추가

GiwanKK committed:
[etc] 각 페이지 장바구니 버튼 추가

GiwanKK committed:
[feat] cart 라우터 추가

GiwanKK committed:
[feat] cart 라우터 추가

GiwanKK committed:
[style] 장밥구니 UI완료

GiwanKK committed:
[style] 장밥구니 UI완료

GiwanKK committed:
[etc] console 오류 제거

GiwanKK committed:
[etc] console 오류 제거

GiwanKK committed:
[feat] sort버튼 클릭 이벤트 정리

GiwanKK committed:
Merge pull request #17 from wecode-bootcamp-korea/component/best

Component/best

syon013 committed:
Merge pull request #18 from wecode-bootcamp-korea/component/product

Component/product

syon013 committed:
Merge pull request #19 from wecode-bootcamp-korea/component/cart

Component/cart

syon013 committed:
Merge branch 'master' into component/main

syon013 committed:
Merge pull request #20 from wecode-bootcamp-korea/component/main

[feat] sort버튼 클릭 이벤트 정리

syon013 committed:
[feat] Main Component EventSwiper 기능 추가

1. 기존 Cover 이미지로 되어 있던 곳을 EventSwiper 기능으로 대체

** 해당 기능을 사용하기 위해서 pull 후 install 필요 **

syon013 committed:
[feat] 로그인, 회원가입 통신 테스트 완료

1. 아이디 저장 기능 사용을 위해 .scss 변경

2. Checked useState 관련 Login -> Form 으로 위치 변경

3. 이메일 선택 사항 -> 필수 사항으로 변경

syon013 committed:
[style] main scss 변경

1. cover 변경하면서 관련 scss도 같이 변경됨.

2. BestList에 버튼 UI 추가(더보기 버튼)
 - 기존에는 div로만 되어 있던 더보기 버튼 Button Component로 변경

GiwanKK committed:
Merge branch 'master' into test/API

GiwanKK committed:
[feat] 제품 리스트 Back연결 test

GiwanKK committed:
[feat] 상세보기 Back 구현 완료

GiwanKK committed:
[etc] css 및 UI 수정

syon013 committed:
[style] 결제하기 페이지 주문고객정보 UI 구현

1. payment Component에서 Modal 창 Open/Close 할 수 있도록 UI 추가

2. 실제 데이터는 기능 개발이 되지 않았으며, UI만 구현해놓은 상태임.

syon013 committed:
[style] payment 오른쪽 영수증 UI Component 구현

1. position 기능을 이용하여 스크롤이 이동되도 영수증은 따라다니도록 구현

2. 아직 주문하기 버튼은 미구현 상태임

syon013 committed:
[style] 배송지 정보/주문 상품 정보 UI 구현

1. payment Component에서 배송지 정보 볼 수 있도록 구현
1-1. 배송지 추가 기능은 미구현 상태

2. 주문상품 UI 구현
2-1. Modal 창 Close 시 상품 최상단 한 개만 보여주고, Open 시 Map으로 가져온 데이터 보여주도록 구현 완료

3. 주문상품에 관련된 이미지 및 목데이터 추가

syon013 committed:
Merge pull request #22 from wecode-bootcamp-korea/component/products

[feat] 제품 리스트 Back연결 test

syon013 committed:
[etc] payment Page 관련 input, button 추가

syon013 committed:
[style] 할인/포인트 Modal Component UI 추가

syon013 committed:
[style] 결제수단 UI Modal Component UI 추가

1. UI 구현

2. onClick 시 마다 on/off 되도록 UI 구현

syon013 committed:
[style] 주문상품 Component Scss 추가

syon013 committed:
[etc] import 위치 변경

syon013 committed:
[style] 구매 동의 UI Component 구현

1. 기능은 미 구현

2. UI Component 만 구현 해놓은 상태임.

syon013 committed:
[style] Payment Component UI 구현 완료

1. 기능 구현 필요

2. 동의 관련 UI까지 구현 완료

GiwanKK committed:
Merge pull request #23 from wecode-bootcamp-korea/component/payment

Component/payment

syon013 committed:
[style] Logo UI 수정

1. Nav Bar Logo UI 수정 및 적용 완료

GiwanKK committed:
[etc] no-image 추가

syon013 committed:
Merge pull request #24 from wecode-bootcamp-korea/component/products

[etc] no-image 추가

syon013 committed:
Merge pull request #21 from wecode-bootcamp-korea/test/API

Test/api

GiwanKK committed:
[feat] 금액 가져오기 수정

syon013 committed:
[etc] package install

syon013 committed:
Merge branch 'master' of https://github.com/wecode-bootcamp-korea/49-2nd-TeaTime-frontend into feat/best

syon013 committed:
[etc] best 기능 추가를 위한 컴포넌트  생성

syon013 committed:
[feat] Nav Bar Link URL 추가

1. Nav Bar에서 베스트, 선물하기 베스트 클릭 시 해당 링크로 이동하도록 상수데이터 수정

syon013 committed:
[style] Main.scss Event Swiper 위치 관련 UI 수정

1. Nav Bar Margin과 Main Padding이 겹쳐 너무 내려가 Main padding  삭제

GiwanKK committed:
[feat] main 제품 가져오기

GiwanKK committed:
[feat] main 제품 상세이동

syon013 committed:
[feat] Login, SignUp API IP 주소 변경

1. 기존에 개인으로 하던 API 주소 AWS AIP 주소로 변경

syon013 committed:
[fix] SignUp Checkbox UI 수정

1. isChecked true 시 나오던 체크 UI가 나오지 않는 현상 관련 버그 수정

syon013 committed:
Merge pull request #25 from wecode-bootcamp-korea/component/products

Component/products

GiwanKK committed:
[fix] 리뷰 리스트 가져오기 오류 수정

GiwanKK committed:
Merge pull request #26 from wecode-bootcamp-korea/feat/best

Feat/best

GiwanKK committed:
[feat] 좋아요 기능 추가 완료

GiwanKK committed:
[etc] main 이번 주 -> 선물하기 베스트

GiwanKK committed:
[style] 별점 수정

GiwanKK committed:
[style] 메인 페딩

GiwanKK committed:
[style] 별점 수정

GiwanKK committed:
[style] 별점 수정

syon013 committed:
Merge pull request #27 from wecode-bootcamp-korea/component/products

[fix] 리뷰 리스트 가져오기 오류 수정

syon013 committed:
[fix] 로그인 아이디 저장 기능 수정

1. 아이디 저장 후 아이디는 Input에 입력되지만, 실제 아이디에 대한 value 값이 들어오지 않는 현상 발생

2. useEffect를 이용하여 아이디 저장 기능이 체크 되어 있는 상태라면 localStorage 에 저장해놓은 Id를 불러와 userInfo에 값을 다시 넣어줌으로써 버그 해결 완료

syon013 committed:
[etc] 이미지 추가 및 scss 수정

1. 위클리 베스트 추가로 인한 banner 이미지 추가

2. tab 관련 scss 변경
 - min-width => max-width 로 변경

syon013 committed:
[feat] weekly Best Chip 추가 및 데이터 추가

1. weekly Best Chip Component 추가

2. weekly 목데이터 추가

3. 상품 목록 정렬을 위하여 쿼리스트링 조건 추가

4. 처음 렌더링 시 데이터를 못 불러오는 현상 방지를 위하여 map함수 사용 전  bestData.length > 0 조건 추가

syon013 committed:
[feat] 배송지 추가 관련 폼 데이터 로직 수정

1. 결제하기 기능에서 유저의 배송지 데이터를 받기 위해서는 mypage의 기능에서 가져오는 수밖엔 없기 때문에 Modal 창에서 유저가 입력한 데이터를 백엔드에서 저장해 받아오는 방식 필요

2. Madal 창에 입력된 value 값을 가져오는 것 까지는 구현 완료, 실제 톹신 테스트 때 추가/입력/수정/삭제 기능 테스트 필요

syon013 committed:
Merge branch 'master' of https://github.com/wecode-bootcamp-korea/49-2nd-TeaTime-frontend

syon013 committed:
Merge branch 'master' of https://github.com/wecode-bootcamp-korea/49-2nd-TeaTime-frontend into feat/best

syon013 committed:
[feat] Nav weekly Best 이동 관련 내용 추가

1. 상수 데이터에 목록 추가

2. Nav Bar Link 수정

syon013 committed:
[feat] 이미지 null로 들어올 시 no-image 파일로 대체 되도록 삼항연산자 추가

syon013 committed:
[feat] Best Page Image Mouse Hover 시 sub Image 띄우도록 기능 구현

1. Image Mouse Hover 시 Sub Image 뜨도록 기능 구현

1-1. isHovering useState 추가

1-2. handelMouseOver, MouseOut 함수 생성

1-3. 이미지가 한 개 일 때는 그냥 setHovering 의 상태를 바꿔주면 되지만, 이미지를 map으로 돌리기 때문에 if문을 추가하여 id와 index 값이 같을 때  해당 이미지만 변경되도록 수정

1-4. onMouseOver, onMouseOut 함수 이용하여 마우스 이벤트 불러오기

GiwanKK committed:
Merge pull request #28 from wecode-bootcamp-korea/feat/best

Feat/best

GiwanKK committed:
Merge branch 'master' into component/cart

syon013 committed:
Merge pull request #29 from wecode-bootcamp-korea/component/cart

Component/cart

GiwanKK committed:
[feat]제품  마우스 오버 이벤트

GiwanKK committed:
[feat]제품  마우스 오버 이벤트

GiwanKK committed:
[style] trbox UI 완료

GiwanKK committed:
[style] trbox UI 완료

GiwanKK committed:
[style] trbox UI 완료

syon013 committed:
Merge pull request #30 from wecode-bootcamp-korea/component/products

[feat]제품  마우스 오버 이벤트

syon013 committed:
[feat] best url 수정

1. best 상품 onClick 시 해당 상품 상세페이지로 이동하도록 링크 수정

syon013 committed:
[etc] payment page 임시 목데이터 생성

syon013 committed:
[feat] 상세페이지 Data , payment page로 옮기기

1. useNavigate() 함수 추가

2. 첫 랜더링 시 카운터 또는 쇼핑백 추가 없이 바로 구매 버튼을 누르면 가격이 들어오지 않는 현상이 발생하여 쇼핑백 관련 useEffect 에 의존성 배열에 'productData' 추가

3. useLocation 을 이용해서 payment Page로 보낼 데이터를 위해 datas 변수 추가

4. handelInfoMove 함수 추가
 - navigate를 이용하여 payment 페이지로 이동하면서 state 값으로 datas 값 가져오도록 로직 구현

 5. 바로구매 버튼을 누르면 4번에서 만든 onClick 함수 작동하도록 추가

syon013 committed:
[feat] Receipt 바로 구매로 가져온 데이터 적용되도록 구현

1. productData props로 데이터를 받아 데이터 보여주도록 구현, toString 및 replace 사용하여 천원 단위로 , 찍도록 구현

syon013 committed:
[feat] itemList 금액 정규표현식 적용

1. price 표시할 때 천원단위로 , 찍히도록 구현

syon013 committed:
[feat] DisCountModal 창에 바로 구매로 가져온 데이터 적용되도록 구현

1. productData를 props로 받아오도록 구현

2. 할인적용 Input에 적용된 할인금액 표시되도록 구현

3. 인풋에 disabled 추가

syon013 committed:
[etc] console.log 삭제

syon013 committed:
[feat] 바로 구매 버튼으로 데이터 가져오도록 기능 구현

1. 상세페이지에서 가져온 데이터가 표시되도록 구현

2. 현재 장바구니는 구현되지 않아 바로 구매 버튼과 관련된 기능만 추가된 상태임.

syon013 committed:
Merge branch 'master' of https://github.com/wecode-bootcamp-korea/49-2nd-TeaTime-frontend

GiwanKK committed:
[style] footer

GiwanKK committed:
[style] main

syon013 committed:
[feat] Scroll Top 기능 추가

1. pathName이 변경될 때마다 스크롤이 최상단으로 이동되도록 컴포넌트 추가

2. 추가한 컴포넌트를 Router에 적용

syon013 committed:
[feat] Nav 오타 수정

syon013 committed:
Merge pull request #31 from wecode-bootcamp-korea/component/products

[style] footer

syon013 committed:
Merge branch 'master' of https://github.com/wecode-bootcamp-korea/49-2nd-TeaTime-frontend into feat/main

syon013 committed:
Merge branch 'master' into feat/payment

syon013 committed:
[style] footer UI 수정

1. footer.scss 수정

2. common.scss 에 로직 추가
 - mypage에 회원정보 탭에 내용이 없는데, 해당 부분에서 footer가 바닥에 안붙고 위로 올라오는 현상 발생

 3. common에서 root 에 display , flex-direction, min-height 속성을 변경함으로써 연장

syon013 committed:
[feat] best 파일에서 div태그 -> main 태그로 변경

GiwanKK committed:
Merge pull request #32 from wecode-bootcamp-korea/feat/payment

Feat/payment

GiwanKK committed:
Merge pull request #33 from wecode-bootcamp-korea/feat/main

Feat/main

GiwanKK committed:
[feat] 장바구니 추가 구현

GiwanKK committed:
[feat] 장바구니 추가 구현

GiwanKK committed:
[feat] 장바구니 추가 구현

GiwanKK committed:
[feat] 장바구니 추가 구현

syon013 committed:
Merge pull request #34 from wecode-bootcamp-korea/component/cart

[feat] 장바구니 추가 구현

syon013 committed:
[etc] Router 수정

syon013 committed:
[etc] Component Style 및 목데이터 수정

syon013 committed:
[feat] best 관련 페이지 정리

1. best.js 안쓰는 import 삭제

2. bestList.js  mouseHover Event 관련 hover 시 구분이 되지 않고 전체적으로 hover 되는 현상 수정

3. bsetList.js Img alt 관련 Wanning 해결

syon013 committed:
[feat] Main 관련 수정 사항

1. Main Chip 관련 버튼 이미지 margin 값 수정

2. Main EventSwiper auto delay 수정 3.5s -> 5s

syon013 committed:
[etc] ComponentData.js 수정

1. 선물추천 메뉴 미 사용으로 인한 삭제

syon013 committed:
[feat] myPage 페이지 수정 관련 사항

1. myPage 회원정보 및 배송지 추가 관련 BackEnd와 통신 / 로직 추가

2. TAP 관련 Component 에 useEffect로 가져온 데이터 props로 전달하도록 로직 구현

3. delivery.js 목데이터 -> BackEnd Data로 수정
 - 처음 랜더링 시 빈 배열로 들어오는 현상으로 인해 map 관련 Error 발생하여 옵셔널 체이닝 사용

4. deliveryAddModal 관련 로직 추가
 4-1. 배송지 추가 버튼 onClick 시 fetch 함수 사용하여 post 요청 보내도록 로직 구현

 4-2. Button Component 스타일링 및 함수 수정

5. DaumPostCode.js console.log 삭제

6. userInfo 하드코딩 -> BackEnd 통신으로 받은 회원정보 Data로 수정

7. deliveryInfo.js 배송지 정보 관련 Input 으로 입력할 수 있도록 UI 수정

8. deliverySelectModal.js 추가
 -  payment Page에서 배송지 수정 버튼 onClick 시 배송지 내용 swap 할 수 있도록 구현 (Input 으로 바꾸면서 기능은 새로 수정 필요)

syon013 committed:
[feat] payment page 수정

1. pront -> pront 로 Data를 받아오던 것을 pront -> back -> pront로 Data를 받아올 수 있도록 수정
 - searchParams 사용

2. 결제 관련 API는 아직 정확한 데이터를 받아오지 못하여 임시 주석 처리

3. payment.scss style 수정

4. Agree.js onClick props 추가

5. receipt.js Data도 backEnd에서 받아온 데이터로 수정

6. product.js 상세구매 페이지에서 결제페이지로 쿼리파라미터 사용
 - 변수 datas는 미사용으로 인한 삭제 처리 예정

 7. trBox 페이지 -> 결제하기 페이지로 Data 넘어가는 것에 대해 확인을 위해 임시 데이터 적용
  - 추후 로직 수정 예정

syon013 committed:
[style] itemList.scss style 수정

syon013 committed:
[etc] 임시 커밋

syon013 committed:
Merge branch 'master' into feat/payment

GiwanKK committed:
Merge pull request #35 from wecode-bootcamp-korea/feat/payment

Feat/payment

GiwanKK committed:
[feat] 장바구니 추가 구현

syon013 committed:
[fix] Login, Signup Api 관련 key 수정

- Snake Case -> Camel Case로 수정

syon013 committed:
[etc] input custom style 추가

syon013 committed:
[feat] myPage 및 payment 기능 추가 중...

syon013 committed:
Merge pull request #36 from wecode-bootcamp-korea/component/cart

[feat] 장바구니 추가 구현

syon013 committed:
Merge branch 'master' of https://github.com/wecode-bootcamp-korea/49-2nd-TeaTime-frontend into feat/payment

syon013 committed:
[feat] Nav 로그아웃 시 로컬스토리지 관련 내용 삭제 로직 수정

1. 로그아웃 시 로컬 스토리지에 있는 모든 내용 삭제 -> 아이디 관련 토큰만 삭제 되도록 로직 수정

syon013 committed:
[etc] 로그인 관련 버그 수정 중

1. 아이디 저장 체크 박스 저장 시 불러온 아이디를 수정할 수 없는 상태. 해당 버그 수정 중

syon013 committed:
[etc] console.log 삭제

- joinForm 관련 내용 console.log 삭제

syon013 committed:
[etc] myPage.js console.log 삭제

syon013 committed:
[etc] Cart.js import 순서 변경

syon013 committed:
[feat] 배송지 입력 관련 UI 수정 및 기능 추가

1. 기존에는 수정이 불가능했던 배송지 관련 기능을 Input 으로 변경

2. BackEed에 Post 관련 Fetch 내용 추가

3. onSubmit 로직 추가

syon013 committed:
[feat] Receipt 데이터 로직 변경

1. Receipt 목업 데이터 -> BackEnd를 통해 받아온 데이터로 로직 수정

2. disabled Scss 추가

syon013 committed:
[feat] Product.js 로직 수정

1. Data 옮기기 위해 만들어두었던 datas 데이터 삭제

GiwanKK committed:
Merge pull request #37 from wecode-bootcamp-korea/feat/payment

Feat/payment

syon013 committed:
[feat] payment 결제하기 Post API 통신 테스트 완료

syon013 committed:
[etc] 프로젝트 README.md 작성

GiwanKK committed:
Merge pull request #38 from wecode-bootcamp-korea/feat/payment

Feat/payment
